app/controllers/admin/content/events.js

```diff
@@ -0,0 +1,95 @@
+import Controller from '@ember/controller';
+
+export default Controller.extend({
+  actions: {
+    async loadSubTopics(topic) {
+      this.set('isLoading', true);
+      try {
+        this.set('subTopics', await topic.get('subTopics'));
+      } catch (e) {
+        this.notify.error(this.get('l10n').t('An unexpected error has occurred. SubTopics not loaded.'));
+      } finally {
+        this.set('isLoading', false);
+      }
+    },
+    openNewEventTypeModal(eventType) {
+      if (eventType) {
+        this.set('eventType', eventType);
+      } else {
+        this.set('eventType', this.store.createRecord('event-type'));
+      }
+      this.set('isNewEventTypeModalOpen', true);
+    },
+    openNewEventTopicModal(eventTopic) {
+      if (eventTopic) {
+        this.set('eventTopic', eventTopic);
+      } else {
+        this.set('eventTopic', this.store.createRecord('event-topic'));
+      }
+      this.set('isNewEventTopicModalOpen', true);
+    },
+    openNewEventSubTopicModal(eventSubTopic) {
+      if (eventSubTopic) {
+        this.set('eventSubTopic', eventSubTopic);
+      } else {
+        this.set('eventSubTopic', this.store.createRecord('event-sub-topic'));
+      }
+      this.set('isNewEventSubTopicModalOpen', true);
+    },
+    deleteEventProperty(eventProp) {
+      this.set('isLoading', true);
+      eventProp.destroyRecord()
+        .then(() => {
+          this.notify.success(this.get('l10n').t('This Event Property has been deleted successfully.'));
+        })
+        .catch(()=> {
+          this.notify.error(this.get('l10n').t('An unexpected error has occurred. Event Type was not deleted.'));
+        })
+        .finally(() => {
+          this.set('isLoading', false);
+        });
+    },
+    addEventType() {
```

Comment:
All these add actions can just be 

```js
addEventProperty(modelInstance) {
  const camelCasedValue = camelCase(modelInstance.modelName);
  this.set('isLoading', true);
  modelInstance.save()
    .then(() => {
      this.notify.success(this.get('l10n').t(`${startCase(camelCasedValue)} has been added successfully.`));
    })
    .catch(() => {
      this.notify.error(this.get('l10n').t(`An unexpected error has occurred. ${startCase(camelCasedValue)} not saved.`));

    })
    .finally(() => {
      this.set(camelCasedValue, null);
      this.set('isLoading', false);
    })

}
```

And you can pass in the `modelInstance` in the template.